adapters/final_characters_controller.go

```diff
@@ -0,0 +1,60 @@
+package adapters
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/SergioSanchezWizeline/academy-go-q12021/helper"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/interfaces"
+	"github.com/SergioSanchezWizeline/academy-go-q12021/usecase"
+)
+
+type finalCharactersController struct {
+	UseCase usecase.FinalCharactersUserCase
+}
+
+func NewFinalCharactersController(usecase usecase.FinalCharactersUserCase) interfaces.Controller {
+	controller := &finalCharactersController{usecase}
+	return controller
+}
+
+func (controller *finalCharactersController) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	opType := GetStringOrDefault(r.URL.Query(), "type", "odd")
+	items := GetIntOrDefault(r.URL.Query(), "items", 10)
+	itemsPerWorker := GetIntOrDefault(r.URL.Query(), "items_per_worker", 10)
+
+	characters, err := controller.UseCase.Execute(opType, items, itemsPerWorker)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "{\"error\": \"%v\"}", err)
+		return
+	}
+
+	json, err := helper.ToJSON(characters)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(w, "{\"error\": \"%v\"}", err)
+		return
+	}
+	fmt.Fprint(w, json)
+}
+
+func GetIntOrDefault(values url.Values, key string, defaul int) int {
+	raw := values.Get(key)
+	if val, ok := strconv.Atoi(raw); ok == nil {
+		return val
+	}
+
+	return defaul
```

Comment:
```suggestion
func GetIntOrDefault(values url.Values, key string, default int) int {
	raw := values.Get(key)
	if val, err := strconv.Atoi(raw); err == nil {
		return val
	}

	return default
```

The `ok == nil` could be a bit confusing, I think it's better to leave it as `err == nil`

Reply:
I changes the error variable, but the `default` variable trigger and error , i think it is because default is a reserved word

Reply:
I see, how about changing it to `defaultValue`?

Reply:
Great idea, ty.

Reply:
Done